framework/midi/internal/synthesizercontroller.cpp

```diff
@@ -17,40 +17,55 @@
 //  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 //=============================================================================
 #include "synthesizercontroller.h"
-
 #include "log.h"
 
 using namespace mu::midi;
 
+SynthesizerController::SynthesizerController()
+    : m_initilizer(nullptr)
+{
+}
+
+SynthesizerController::~SynthesizerController()
+{
+    if (m_initilizer && m_initilizer->joinable()) {
+        m_initilizer->join();
+    }
+}
+
 void SynthesizerController::init()
 {
     IF_ASSERT_FAILED(audioEngine()) {
         return;
     }
 
-    auto init = [this](float sampleRate) {
-                    std::vector<std::shared_ptr<ISynthesizer> > synthesizers = synthRegister()->synthesizers();
-
-                    for (std::shared_ptr<ISynthesizer> synth : synthesizers) {
-                        synth->init(sampleRate);
-
-                        reloadSoundFonts(synth);
-                        auto notification = sfprovider()->soundFontPathsForSynthChanged(synth->name());
-                        notification.onNotify(this, [this, synth]() { reloadSoundFonts(synth); });
-                    }
-                };
-
     if (audioEngine()->isInited()) {
-        init(audioEngine()->sampleRate());
+        initSoundFonts(audioEngine()->sampleRate());
     } else {
-        audioEngine()->initChanged().onReceive(this, [this, init](bool inited) {
+        audioEngine()->initChanged().onReceive(this, [this](bool inited) {
             if (inited) {
-                init(audioEngine()->sampleRate());
+                initSoundFonts(audioEngine()->sampleRate());
             }
         });
     }
 }
 
+void SynthesizerController::initSoundFonts(unsigned int sampleRate)
+{
+    m_initilizer = std::make_shared<std::thread>([=]() {
+        std::vector<std::shared_ptr<ISynthesizer> > synthesizers = synthRegister()->synthesizers();
+
+        for (std::shared_ptr<ISynthesizer> synth : synthesizers) {
+            synth->init(sampleRate);
```

Comment:
I added 
```
Ret FluidSynth::init(float samplerate)
{
    static std::atomic_int call_count = 0;
    ++call_count;
    LOGI() << "begin: th: " << mu::runtime::threadName() << ", call count: " << call_count << ", samplerate: " << samplerate;
...
    LOGI() << "end: th: " << mu::runtime::threadName() << ", call count: " << call_count;
```

Output
```
02:01:23.343 | INFO  | main_thread | FluidSynth | init: begin: th: main, call count: 1, samplerate: 48000
02:01:23.347 | INFO  | main_thread | FluidSynth | init: end: th: main, call count: 1
02:01:23.347 | INFO  | main_thread | FluidSynth | init: begin: th: main, call count: 2, samplerate: 1
02:01:23.348 | ERROR | main_thread | FluidSynth | init: requested set value for 'synth.sample-rate' out of range
02:01:23.349 | INFO  | main_thread | FluidSynth | init: end: th: main, call count: 2
02:01:23.635 | INFO  | main_thread | FluidSynth | init: begin: th: main, call count: 3, samplerate: 48000
02:01:23.636 | INFO  | main_thread | FluidSynth | init: end: th: main, call count: 3
02:01:23.639 | INFO  | 139939795715840 | FluidSynth | init: begin: th: 139939795715840, call count: 4, samplerate: 48000
02:01:23.641 | INFO  | 139939795715840 | FluidSynth | init: end: th: 139939795715840, call count: 4
```

Who is responsible for initializing synthesizers?
Why is it necessary to call initialization 4 times?

It also leads to a memory leak, because the init is `new`...